packages/charting/src/components/Legends/Legends.base.tsx

```diff
@@ -238,12 +241,20 @@ export class LegendsBase extends React.Component<ILegendsProps, ILegendState> {
         onBlur={onMouseOut}
         data-is-focusable={true}
       >
-        <div className={classNames.rect} />
+        <div className={this._getShapeClass(classNames, legend)} />
         <div className={classNames.text}>{legend.title}</div>
       </div>
     );
   };
 
+  private _getShapeClass(classNames: IProcessedStyleSet<ILegendsStyles>, legend: ILegend): string {
```

Comment:
I think this should also be in the styles function. We can just return a shape. The styles function can handle the logic dictating what shape to return.

Reply:
I don't quite understand how to change it. Based on my understanding is just to move this function in the styles file.
It is helpful if you can show me the code.

Reply:
The thought was that instead of having a string change which class is applied, change how the style actually renders and have one less style area to deal with.